### DISCUSSION TITLE
Cellular connection with Quectel EC25 LTE modem USB on BBN/lysmarine

### DISCUSSION THREAD

**Discussion by atchik6056** [bareboat-necessities/lysmarine_gen#344] [Q&A] [answered]
Hello
Having problem to setup Quectel Wireless Solutions Co., Ltd. EC25 LTE modem on BBN/lysmarine 2023-07-23.
it is Alfa-tube-u 4g modem USB attached and discovered by BBN as 
Bus 001 Device 012: ID 2c7c:0125 Quectel Wireless Solutions Co., Ltd. EC25 LTE modem
I am following procedure described at: https://www.jeffgeerling.com/blog/2022/using-4g-lte-wireless-modems-on-raspberry-pi, which reports that EC25 LTE modem should work OK on raspi. 
I also contacted Quectel support and they confirmed that device driver is built into kernel of this version 
Linux xxx 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr 3 17:24:16 BST 2023 aarch64 GNU/Linux
When I plugin device to USB, BBN discovers it ok, and creates /dev/ttyUSB2...4 as described in the above article.
However BBN failed to create /dev/cdc-wdm0 so I stuck here.
ECM method does not work as well, minicon does not do anything although it show /dev/ttyUSB2 device is connected ok.
I also tried this workaround to reload qmi-wwan first and the option the second, but it did not helped: https://raspberrypi.stackexchange.com/questions/68978/loading-kernel-module-qmi-wwan-before-option-to-get-dev-cdc-wdm0-automatica

Any suggestion or sharing experience or hints is very appreciated.

PS:  Alfa-tube-u 4g modem works and provides very strong and reliable connection with ubuntu 20.04 LTS or 22.04 LTS.
PS2: openplotter v3 does not work as well as BBN, same diagnostics.

**Comment by mgrouch** [comment]
If it works on 22.04 LTS Ubuntu you should be
able to make it work under BBN which is bullseye Debian based just as Ubuntu. 

/dev/cdc* device

is that a symbolic link on Ubuntu where it works? If it is you might be missing udev rule. 

If it is not symbolic link you might be missing some driver which needs to be loaded
/etc/modules

You can check with lsof command if some other processes open your device and make a conflict. 

The vendor of your device should provide you with support why it’s not working. At least they have that device

**Reply by atchik6056** [reply]
I have contacted quectel, in dialog so far
a couple of things I noticed compare to ubuntu: it does not load usbnet and cdc-ether, possibly root cause is cdc-ether as it loads usbnet on ubuntu.
dmesg:

````

[ 2590.726789] usb 1-1.1.1: new high-speed USB device number 7 using xhci_hcd
[ 2590.884905] usb 1-1.1.1: New USB device found, idVendor=2c7c, idProduct=0125, bcdDevice= 3.18
[ 2590.884925] usb 1-1.1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[ 2590.884931] usb 1-1.1.1: Product: EG25-G
[ 2590.884937] usb 1-1.1.1: Manufacturer: Quectel
[ 2590.891512] option 1-1.1.1:1.0: GSM modem (1-port) converter detected
[ 2590.891771] usb 1-1.1.1: GSM modem (1-port) converter now attached to ttyUSB1
[ 2590.892003] option 1-1.1.1:1.1: GSM modem (1-port) converter detected
[ 2590.892183] usb 1-1.1.1: GSM modem (1-port) converter now attached to ttyUSB2
[ 2590.892409] option 1-1.1.1:1.2: GSM modem (1-port) converter detected
[ 2590.892573] usb 1-1.1.1: GSM modem (1-port) converter now attached to ttyUSB3
[ 2590.892788] option 1-1.1.1:1.3: GSM modem (1-port) converter detected
[ 2590.892936] usb 1-1.1.1: GSM modem (1-port) converter now attached to ttyUSB4
[ 2590.899250] cdc_ether: probe of 1-1.1.1:1.4 failed with error -22

````

**Reply by mgrouch** [reply]
BBN OS is not based on Ubuntu. 
BBN OS is based on Raspberry Pi OS bullseye. 

Can you check with your hardware vendor how to set up your device on  Raspberry Pi OS bullseye?

**Reply by atchik6056** [reply, marked as answer]
I have managed to connect to the cellular network, the problem was (from Quectel guy [jfrog](https://forums.quectel.com/u/jfrog)- very good and responsive): 

`Your modem is in ECM mode but cdc_ether driver failed to probe the device.
If you want using QMI you need to switch the modem using AT+QCFG="usbnet",0 (through /dev/ttyUSB3)
`


as I was playing with ECM mode previously
another thing was: qmicli does not send PIN, or I do not know how to do it, so I switched to ModemManager and connected to the network successfully. ModemManager is very well documented, so something like:

````
sudo systemctl status ModemManager
sudo systemctl start ModemManager
sudo mmcli -L
mmcli -m 0 -K | grep "modem.generic.sim" | awk -F ": " '{ print $2 }'
sudo mmcli -i 0 --pin=1234
sudo mmcli -m 0 --simple-connect="pin=1234,apn=internet"
````

...
now in order to successfully run udhcpc you need to set raw_ip:

````

sudo ip link set wwan0 down
 echo 'Y' | sudo tee /sys/class/net/wwan0/qmi/raw_ip
Y
sudo ip link set wwan0 up
sudo udhcpc -q -f -i wwan0

````

...
now I need to setup proper routing 
...
PS: after you send PIN with mmcli command you can use qmicli as here: https://www.jeffgeerling.com/blog/2022/using-4g-lte-wireless-modems-on-raspberry-pi

**Reply by mgrouch** [reply]
I’m glad you solved that part. Thanks. 
I’ve edited your comment to mark code
snippets easier to copy for others.

**Reply by atchik6056** [reply]
in fact, if you install modem-manager-gui then it gives you the same look and feel as on ubuntu 20.04 lts, though i cannot find field to enter PIN there. However it shows traffic, sms, and so on. I am not sure on the future of GUI because for example in Ubuntu 22.04 it is already broken by "keen" developers.
PS: I couldn't make cellular connection on openplotter usable due their network implementation specifics (bridge), BBN seems more standard to debian/ubuntu so more friendly, just my 2 cents, thank you Michael for your work!
